lib/active_merchant/billing/gateways/globalcollect.rb

```diff
@@ -0,0 +1,294 @@
+        add_customer_data(post, options)
+        commit(:capture, post, authorization)
+      end
+
+      def refund(money, authorization, options={})
+        post = nestable_hash
+        add_amount(post, money, options={})
+        add_refund_customer_data(post, options)
+        commit(:refund, post, authorization)
+      end
+
+      def void(authorization, options={})
+        post = nestable_hash
+        commit(:void, post, authorization)
+      end
+
+      def verify(payment, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, payment, options) }
+          r.process { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: )[^\\]*)i, '\1[FILTERED]').
+          gsub(%r(("cardNumber\\":\\")\d+), '\1[FILTERED]').
+          gsub(%r(("cvv\\":\\")\d+), '\1[FILTERED]')
+      end
+
+      private
+
+      BRAND_MAP = {
+        "visa" => "1",
+        "american_express" => "2",
+        "master" => "3",
+        "discover" => "128"
+      }
+
+      def add_order(post, money, options)
+        post["order"]["amountOfMoney"] = {
+          "amount" => amount(money),
+          "currencyCode" => options[:currency] || currency(money)
+        }
+        post["order"]["references"] = {
+          "merchantOrderId" => options[:order_id], # Max 10 digits
+          "merchantReference" => options[:order_id],
+          "descriptor" => options[:description] # Max 256 chars
+        }
+        if options[:invoicedata]
+          post["order"]["reference"]["invoiceData"] = {
+            "invoiceNumber" => options[:invoicenumber]
+          }
+        end
+      end
+
+      def add_amount(post, money, options)
+        post["amountOfMoney"] = {
+          "amount" => amount(money),
+          "currencyCode" => options[:currency] || currency(money)
+        }
+      end
+
+      def add_payment(post, payment)
+        year  = format(payment.year, :two_digits)
+        month = format(payment.month, :two_digits)
+        expirydate =   "#{month}#{year}"
+
+        post["cardPaymentMethodSpecificInput"] = {
+            "paymentProductId" => BRAND_MAP[payment.brand],
+            "skipAuthentication" => "true", # refers to 3DSecure
+            "skipFraudService" => "true"
+        }
+        post["cardPaymentMethodSpecificInput"]["card"] = {
+            "cvv" => payment.verification_value,
+            "cardNumber" => payment.number,
+            "expiryDate" => expirydate,
+            "cardholderName" => payment.name
+        }
+      end
+
+      def add_customer_data(post, options)
+        post["order"]["customer"] = {
+          "merchantCustomerId" => options[:customer]
+        }
+        post["order"]["customer"]["personalInformation"] = {
+          "name" => {
+            "firstName" => options[:firstname],
```

Comment:
We should get the first/last names [from the payment method](https://github.com/activemerchant/active_merchant/blob/b1c5c11812a2284a2a9c566e5efd34064006040b/lib/active_merchant/billing/credit_card.rb#L117-L122) instead of having the merchant pass in data in the options we already have.